src/SOS/Strike/strike.cpp

```diff
@@ -2464,7 +2466,27 @@ size_t FormatGeneratedException (DWORD_PTR dataPtr,
             WCHAR filename[MAX_LONGPATH] = W("");
             ULONG linenum = 0;
             if (bLineNumbers && 
-                SUCCEEDED(GetLineByOffset(TO_CDADDR(ste.ip), &linenum, filename, _countof(filename))))
+                // To get the source line number of the actual code that threw an exception, the IP needs 
+                // to be adjusted in certain cases. 
+                //
+                // The IP of the leaf node stack frame (i == 0) points to either:
+                //
+                // 1) The instruction that caused a hardware exception (div by zero, null ref, etc).
+                // 2) After the call instruction to an internal runtime function (FCALL like IL_Throw,
+                //    IL_Rethrow, JIT_OverFlow, etc.) that caused a software exception.
+                //
+                // It is case 2 that needs to adjust IP because these IL calls don't contain any other instructions
+                // after and point to next IL instruction and source line after the throw. We distinguish between #1
+                // and #2 by the bAsync flag which is set to true for hardware exceptions.
+                //
+                // The rest of the stack frame IPs also point to the instruction after the call. The difference 
+                // in this case is that these IL calls consist of more code after the call (at the minimum a nop
+                // sequence point) that is consider part of the IL instruction. The IP doesn't need to be adjusted
+                // in these cases.
+                //
+                // When the IP needs to be adjusted it is a lot simpler to decrement IP instead of trying to figure
+                // out the beginning of the instruction. It is enough for GetLineByOffset to return the correct line number.
```

Comment:
I would also include some text that says we are choosing to print the unmodified IP (which points after the exception causing instruction in most cases). This means that the printed IP and the printed line number often will not map to one another, and this is intentional. We expect people reading the IP portion of the output to understand how to interpret it.